pkg/controllers/matrixone/handler.go

```diff
@@ -210,48 +191,43 @@ func sdkCreateOrUpdateAsNeeded(
 	if obj.GetAnnotations()[matrixoneOpResourceHash] != prevObj.GetAnnotations()[matrixoneOpResourceHash] || !isEqualFn(prevObj, obj) {
 		obj.SetResourceVersion(prevObj.GetResourceVersion())
 		updaterFn(prevObj, obj)
-		update, err := writers.Update(context.TODO(), sdk, moc, obj, emitEvent)
+		update, err := Update(context.TODO(), sdk, moc, obj, emitEvent)
 		if err != nil {
 			return "", err
 		}
 		return update, err
 
 	}
 	return "", nil
-
 }
 
-func isObjFullyDeployed(
-	sdk client.Client,
-	moc *v1alpha1.MatrixoneCluster,
-	emptyObjFn func() object,
-	emitEvent EventEmitter) (bool, error) {
-
-	// Get Object
-	obj, err := readers.Get(context.TODO(), sdk, moc, emptyObjFn, emitEvent)
+func isStatefulsetFullyDeployed(sdk client.Client, moc *v1alpha1.MatrixoneCluster, emitEvent EventEmitter) (bool, error) {
```

Comment:
In my view, object is basic concept, So we do something for it. You can segregate check functions logic, but keep a entry function for objects :-)

Reply:
> I found it is okay to segregate this check function for different workloads since the check logic has little common logic. How do you like? @wanglei4687

In my view, object is basic concept, So we do something for it. You can segregate check functions logic, but keep a entry function for objects :-)

Reply:
sounds good, addressed in https://github.com/matrixorigin/matrixone-operator/pull/114/commits/613b2ddefecb8c86ad48f8138c19922a63ccab36